diff --git a/quarkus/addons/kubernetes/deployment/pom.xml b/quarkus/addons/kubernetes/deployment/pom.xml
index c265505adf..453ef2cdd2 100644
--- a/quarkus/addons/kubernetes/deployment/pom.xml
+++ b/quarkus/addons/kubernetes/deployment/pom.xml
@@ -56,6 +56,16 @@
           </annotationProcessorPaths>
         </configuration>
       </plugin>
+      <!-- Needed by QuarkusUnitTest class. JBoss logging must be in bootstrap logger-->
+      <plugin>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-extension-maven-plugin</artifactId>
+        <configuration>
+            <parentFirstArtifacts>
+                <parentFirstArtifact>org.jboss.logging:jboss-logging</parentFirstArtifact>
+            </parentFirstArtifacts>
+        </configuration>
+    </plugin>
     </plugins>
   </build>
 </project>
diff --git a/quarkus/addons/kubernetes/deployment/src/test/java/org/kie/kogito/addons/quarkus/kubernetes/KubernetesAddOnTest.java b/quarkus/addons/kubernetes/deployment/src/test/java/org/kie/kogito/addons/quarkus/kubernetes/KubernetesAddOnTest.java
index 77a4491cbd..b93abb6fa5 100644
--- a/quarkus/addons/kubernetes/deployment/src/test/java/org/kie/kogito/addons/quarkus/kubernetes/KubernetesAddOnTest.java
+++ b/quarkus/addons/kubernetes/deployment/src/test/java/org/kie/kogito/addons/quarkus/kubernetes/KubernetesAddOnTest.java
@@ -15,8 +15,6 @@
  */
 package org.kie.kogito.addons.quarkus.kubernetes;
 
-import jakarta.inject.Inject;
-
 import org.jboss.shrinkwrap.api.ShrinkWrap;
 import org.jboss.shrinkwrap.api.spec.JavaArchive;
 import org.junit.jupiter.api.Test;
@@ -26,6 +24,8 @@ import org.kie.kogito.addons.k8s.workitems.AbstractDiscoveredEndpointCaller;
 
 import io.quarkus.test.QuarkusUnitTest;
 
+import jakarta.inject.Inject;
+
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
 public class KubernetesAddOnTest {
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointCallerProducer.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointCallerProducer.java
index 2ee0bfca5d..1bfda3ee04 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointCallerProducer.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointCallerProducer.java
@@ -15,17 +15,17 @@
  */
 package org.kie.kogito.addons.quarkus.k8s;
 
+import org.kie.kogito.addons.k8s.EndpointDiscovery;
+import org.kie.kogito.addons.quarkus.k8s.workitems.QuarkusDiscoveredEndpointCaller;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.inject.Default;
 import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Inject;
 import jakarta.inject.Singleton;
 
-import org.kie.kogito.addons.k8s.EndpointDiscovery;
-import org.kie.kogito.addons.quarkus.k8s.workitems.QuarkusDiscoveredEndpointCaller;
-
-import com.fasterxml.jackson.databind.ObjectMapper;
-
 @ApplicationScoped
 public class EndpointCallerProducer {
 
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointDiscoveryProducer.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointDiscoveryProducer.java
index 4471230cf3..b19b9e5194 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointDiscoveryProducer.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/EndpointDiscoveryProducer.java
@@ -15,6 +15,12 @@
  */
 package org.kie.kogito.addons.quarkus.k8s;
 
+import org.kie.kogito.addons.k8s.KnativeRouteEndpointDiscovery;
+import org.kie.kogito.addons.k8s.KubernetesServiceEndpointDiscovery;
+import org.kie.kogito.addons.k8s.ServiceAndThenRouteEndpointDiscovery;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.inject.Default;
 import jakarta.enterprise.inject.Produces;
@@ -22,12 +28,6 @@ import jakarta.inject.Inject;
 import jakarta.inject.Named;
 import jakarta.inject.Singleton;
 
-import org.kie.kogito.addons.k8s.KnativeRouteEndpointDiscovery;
-import org.kie.kogito.addons.k8s.KubernetesServiceEndpointDiscovery;
-import org.kie.kogito.addons.k8s.ServiceAndThenRouteEndpointDiscovery;
-
-import io.fabric8.kubernetes.client.KubernetesClient;
-
 @ApplicationScoped
 public class EndpointDiscoveryProducer {
 
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeClientProducer.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeClientProducer.java
index fce0eb001e..46142bfac9 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeClientProducer.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeClientProducer.java
@@ -15,13 +15,13 @@
  */
 package org.kie.kogito.addons.quarkus.k8s.discovery;
 
+import io.fabric8.knative.client.KnativeClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.inject.Produces;
 import jakarta.inject.Inject;
 
-import io.fabric8.knative.client.KnativeClient;
-import io.fabric8.kubernetes.client.KubernetesClient;
-
 final class KnativeClientProducer {
 
     private final KubernetesClient kubernetesClient;
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscovery.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscovery.java
index a419e5ac98..72419eb4f2 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscovery.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscovery.java
@@ -18,15 +18,15 @@ package org.kie.kogito.addons.quarkus.k8s.discovery;
 import java.net.URI;
 import java.util.Optional;
 
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.inject.Inject;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import io.fabric8.knative.client.KnativeClient;
 import io.fabric8.knative.serving.v1.Service;
 
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
 @ApplicationScoped
 public class KnativeServiceDiscovery extends AbstractResourceDiscovery {
 
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftResourceDiscovery.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftResourceDiscovery.java
index 12a1f156ec..005858c320 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftResourceDiscovery.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftResourceDiscovery.java
@@ -20,9 +20,6 @@ import java.net.URI;
 import java.util.Optional;
 import java.util.function.Supplier;
 
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.inject.Inject;
-
 import org.kie.kogito.addons.quarkus.k8s.KubeConstants;
 import org.kie.kogito.addons.quarkus.k8s.discovery.utils.PodUtils;
 import org.kie.kogito.addons.quarkus.k8s.discovery.utils.ServiceUtils;
@@ -34,6 +31,9 @@ import io.fabric8.openshift.api.model.DeploymentConfig;
 import io.fabric8.openshift.api.model.Route;
 import io.fabric8.openshift.client.OpenShiftClient;
 
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
 @ApplicationScoped
 public class OpenShiftResourceDiscovery extends AbstractResourceDiscovery {
 
diff --git a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/VanillaKubernetesResourceDiscovery.java b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/VanillaKubernetesResourceDiscovery.java
index a7f7ac1619..dea6aa2ff9 100644
--- a/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/VanillaKubernetesResourceDiscovery.java
+++ b/quarkus/addons/kubernetes/runtime/src/main/java/org/kie/kogito/addons/quarkus/k8s/discovery/VanillaKubernetesResourceDiscovery.java
@@ -19,9 +19,6 @@ import java.net.URI;
 import java.util.Optional;
 import java.util.function.Supplier;
 
-import jakarta.enterprise.context.ApplicationScoped;
-import jakarta.inject.Inject;
-
 import org.kie.kogito.addons.quarkus.k8s.discovery.utils.DeploymentUtils;
 import org.kie.kogito.addons.quarkus.k8s.discovery.utils.IngressUtils;
 import org.kie.kogito.addons.quarkus.k8s.discovery.utils.PodUtils;
@@ -32,6 +29,9 @@ import org.slf4j.LoggerFactory;
 
 import io.fabric8.kubernetes.client.KubernetesClient;
 
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+
 @ApplicationScoped
 public class VanillaKubernetesResourceDiscovery extends AbstractResourceDiscovery {
 
diff --git a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KnativeRouteEndpointDiscoveryTest.java b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KnativeRouteEndpointDiscoveryTest.java
index ff8268bd65..d2408b7659 100644
--- a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KnativeRouteEndpointDiscoveryTest.java
+++ b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KnativeRouteEndpointDiscoveryTest.java
@@ -22,8 +22,6 @@ import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
-import jakarta.inject.Inject;
-
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.addons.k8s.Endpoint;
@@ -37,6 +35,8 @@ import io.fabric8.kubernetes.client.KubernetesClient;
 import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.kubernetes.client.WithKubernetesTestServer;
 
+import jakarta.inject.Inject;
+
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
diff --git a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KubernetesServiceEndpointDiscoveryTest.java b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KubernetesServiceEndpointDiscoveryTest.java
index 1e9cabca8c..d5be5ce3af 100644
--- a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KubernetesServiceEndpointDiscoveryTest.java
+++ b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/KubernetesServiceEndpointDiscoveryTest.java
@@ -23,9 +23,6 @@ import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 
-import jakarta.inject.Inject;
-import jakarta.inject.Named;
-
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.addons.k8s.Endpoint;
 import org.kie.kogito.addons.k8s.EndpointDiscovery;
@@ -39,6 +36,9 @@ import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.kubernetes.client.KubernetesTestServer;
 import io.quarkus.test.kubernetes.client.WithKubernetesTestServer;
 
+import jakarta.inject.Inject;
+import jakarta.inject.Named;
+
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.Assertions.fail;
diff --git a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscoveryTest.java b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscoveryTest.java
index 84fc2e3971..5957cf2036 100644
--- a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscoveryTest.java
+++ b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/KnativeServiceDiscoveryTest.java
@@ -18,8 +18,6 @@ package org.kie.kogito.addons.quarkus.k8s.discovery;
 import java.net.URI;
 import java.util.Optional;
 
-import jakarta.inject.Inject;
-
 import org.junit.jupiter.api.Test;
 import org.kie.kogito.addons.quarkus.k8s.test.utils.KnativeResourceDiscoveryTestUtil;
 
@@ -28,6 +26,8 @@ import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.kubernetes.client.KubernetesTestServer;
 import io.quarkus.test.kubernetes.client.WithKubernetesTestServer;
 
+import jakarta.inject.Inject;
+
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 
 @QuarkusTest
diff --git a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftServiceDiscoveryTest.java b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftServiceDiscoveryTest.java
index 8304a33f66..b2b6715b48 100644
--- a/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftServiceDiscoveryTest.java
+++ b/quarkus/addons/kubernetes/runtime/src/test/java/org/kie/kogito/addons/quarkus/k8s/discovery/OpenShiftServiceDiscoveryTest.java
@@ -18,8 +18,6 @@ package org.kie.kogito.addons.quarkus.k8s.discovery;
 import java.net.URI;
 import java.util.Optional;
 
-import jakarta.inject.Inject;
-
 import org.junit.jupiter.api.Test;
 
 import io.fabric8.kubernetes.api.model.Pod;
@@ -32,6 +30,8 @@ import io.quarkus.test.junit.QuarkusTest;
 import io.quarkus.test.kubernetes.client.OpenShiftTestServer;
 import io.quarkus.test.kubernetes.client.WithOpenShiftTestServer;
 
+import jakarta.inject.Inject;
+
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 @QuarkusTest
diff --git a/quarkus/addons/kubernetes/test-utils/src/main/java/org/kie/kogito/addons/quarkus/k8s/test/utils/KnativeResourceDiscoveryTestUtil.java b/quarkus/addons/kubernetes/test-utils/src/main/java/org/kie/kogito/addons/quarkus/k8s/test/utils/KnativeResourceDiscoveryTestUtil.java
index 5bcafe29dc..eadc5283bd 100644
--- a/quarkus/addons/kubernetes/test-utils/src/main/java/org/kie/kogito/addons/quarkus/k8s/test/utils/KnativeResourceDiscoveryTestUtil.java
+++ b/quarkus/addons/kubernetes/test-utils/src/main/java/org/kie/kogito/addons/quarkus/k8s/test/utils/KnativeResourceDiscoveryTestUtil.java
@@ -33,7 +33,7 @@ public final class KnativeResourceDiscoveryTestUtil {
             Service service = knativeClient.services()
                     .inNamespace(namespace)
                     .load(getResourceAsStream(knativeYaml))
-                    .get();
+                    .item();
 
             service.getStatus().setUrl(remoteServiceUrl);
 
